packages/core/src/middleware/koa-oidc-error-handler.test.ts

```diff
@@ -1,118 +1,98 @@
+    await expect(koaOidcErrorHandler()(ctx, next)).rejects.toMatchError(error);
   });
 
-  it('Invalid Scope', async () => {
+  it('should recognize invalid scope error', async () => {
+    const ctx = createContextWithRouteParameters();
     const error_description = 'Mock scope is invalid';
     const mockScope = 'read:foo';
-    const error = new errors.InvalidScope(error_description, mockScope);
-    next.mockImplementationOnce(() => {
-      throw error;
-    });
-
-    await expect(koaOIDCErrorHandler()(ctx, next)).rejects.toMatchError(
-      new RequestError(
-        {
-          code: 'oidc.invalid_scope',
-          status: error.status,
-          expose: true,
-          scope: mockScope,
-        },
-        { error_description }
-      )
-    );
+    await expectErrorResponse(ctx, new errors.InvalidScope(error_description, mockScope));
   });
 
-  it('Session Not Found', async () => {
-    const error_description = 'session not found';
-    const error = new errors.SessionNotFound('session not found');
+  it('should transform session not found error code', async () => {
+    const ctx = createContextWithRouteParameters();
 
-    next.mockImplementationOnce(() => {
-      throw error;
+    await expectErrorResponse(ctx, new errors.SessionNotFound('session not found'), {
+      code: 'session.not_found',
+      message: i18next.t('errors:session.not_found'),
     });
-
-    await expect(koaOIDCErrorHandler()(ctx, next)).rejects.toMatchError(
-      new RequestError(
-        {
-          code: 'session.not_found',
-          status: error.status,
-          expose: true,
-        },
-        {
-          error_description,
-        }
-      )
-    );
   });
 
-  it('Insufficient Scope', async () => {
+  it('should add scope in response when needed', async () => {
+    const ctx = createContextWithRouteParameters();
     const error_description = 'Insufficient scope for access_token';
     const scope = 'read:foo';
 
-    const error = new errors.InsufficientScope(error_description, scope);
-
-    next.mockImplementationOnce(() => {
-      throw error;
+    await expectErrorResponse(ctx, new errors.InsufficientScope(error_description, scope), {
+      scope,
     });
-
-    await expect(koaOIDCErrorHandler()(ctx, next)).rejects.toMatchError(
-      new RequestError(
-        {
-          code: 'oidc.insufficient_scope',
-          status: error.status,
-          expose: true,
-          scopes: scope,
-        },
-        {
-          error_description,
-        }
-      )
-    );
   });
 
-  it('Unhandled OIDCProvider Error', async () => {
-    const error = new errors.AuthorizationPending();
+  it('should add error uri when available', async () => {
+    const ctx = createContextWithRouteParameters();
+    const error = new errors.InvalidGrant('invalid grant');
 
-    next.mockImplementationOnce(() => {
-      throw error;
+    await expectErrorResponse(ctx, error, {
+      error_uri: 'https://openid.sh/debug/invalid_grant',
```

Comment:
nit: should maintain a constant list for this and upcoming URLs.

Reply:
do you mean [this](https://github.com/logto-io/logto/pull/4573/files#diff-b706e9a1b8a197003ab8a28985e03df08134d5830706d3fccf17fdca3b842373R11-R15)?

Reply:
Oh, yes.